### Formler og følger

I matematikken er formler sammenhenger som blir uttrykt med matematisk språk. Vi bruker ofte formler til å regne på ulike størrelser i virkeligheten og i geometrien. Formler er også viktige når vi snakker om figurtall.

```{admonition} Forutsetninger
:class: note
Denne siden forutsetter at man har vært gjennom delkapitlene
- {ref}`Variabler og syntaks`
- {ref}`Kalkulatorer og pakker`
- {ref}`Vilkår`
- {ref}`Løkker`
- {ref}`Lister`
```

#### Eksempler på formler.
Her kommer noen eksempler på formler fra geometrien, fysikk og naturfag.

##### Plangeometri
::::{grid} 3
:gutter: 1

:::{grid-item-card} Navn
:::

:::{grid-item-card} Areal
:::

:::{grid-item-card} Omkrets

:::

:::{grid-item-card}
Sirkel
:::

:::{grid-item-card}
$\pi r^2$
:::

:::{grid-item-card}
$2\pi r$
:::

:::{grid-item-card}
Trekant
:::

:::{grid-item-card}
$\frac{1}{2}gh$
:::

:::{grid-item-card}
$a+b+c$
:::

:::{grid-item-card}
Rektangel
:::

:::{grid-item-card}
$a\cdot b$
:::

:::{grid-item-card}
$2a + 2b$
:::

::::

##### Romgeometri 
::::{grid} 3
:gutter: 1

:::{grid-item-card} Navn
:::

:::{grid-item-card} Volum
:::

:::{grid-item-card} Overflate

:::

:::{grid-item-card}
Kule
:::

:::{grid-item-card}
$\frac{4}{3}\pi r^3$
:::

:::{grid-item-card}
$4\pi r^2$
:::

:::{grid-item-card}
Sylinder
:::

:::{grid-item-card}
$\pi r^2h$
:::

:::{grid-item-card}
$2\pi r^2 + 2\pi rh$
:::

:::{grid-item-card}
Kjegle
:::

:::{grid-item-card}
$\frac{1}{3}\pi r^2h$
:::

:::{grid-item-card}
$\pi r^2 + \pi rs$
:::

::::

Ved å bruke litt algebra kan vi gjøre om på formlene til å finne andre størrelser enn de som er oppgitt i formelen.

#### Tallfølger

Her kommer noen vanlige tallfølger. Formlene under er gitt for tallene $n\in \{1, 2, 3, \dots\}$.

::::{grid} 3
:gutter: 1

:::{grid-item-card} Navn
:::

:::{grid-item-card} Figur
:::

:::{grid-item-card} Formel
:::


:::{grid-item-card}
De naturlige tallene
:::

:::{grid-item-card}

:::

:::{grid-item-card}
$n$
:::


:::{grid-item-card}
Kvadrattallene
:::

:::{grid-item-card}

:::

:::{grid-item-card}
$n^2$
:::


:::{grid-item-card}
Rektangeltallene
:::

:::{grid-item-card}

:::

:::{grid-item-card}
$n(n+1)$
:::


:::{grid-item-card}
Trekanttallene
:::

:::{grid-item-card}

:::

:::{grid-item-card}
$\frac{n(n+1)}{2}$
:::

::::

##### Programmere tallfølger med formel

Vi kan enkelt programmere tallfølger som har formel. Da kan vi lett finne ulike summer og identifisere hvilke tall som er i følgen, og hvilke som ikke er det.

La oss finne summen av de $20$ første kvadrattallene og undersøke om tallet $289$ er et kvadrattall.

In [3]:
s = 0 # Summen starter som null

for n in range(1, 21): # For-løkken går fra 1 til én mindre enn den siste parameteren, altså 20.
    s += n**2          # n**2 er formelen.
    print(n, n**2)     # Skriver ut tallene

print(f"Summen er {s}")

1 1
2 4
3 9
4 16
5 25
6 36
7 49
8 64
9 81
10 100
11 121
12 144
13 169
14 196
15 225
16 256
17 289
18 324
19 361
20 400
Summen er 2870


Som vi ser er tallet $289$ et kvadrattall, det er nummer 17 i følgen. I tillegg er summen av de $20$ første kvadrattallene lik $2870$.

##### Programmere tallfølger med algoritme

Det er ikke like lett å finne en formel for alle tallfølger. Noen ganger er det enklere å programmere tallfølgen fra en algoritme.

La oss se på *fibonaccitallene*. De er gitt ved denne algoritmen.

- Start med tallene $\{0, 1\}$
- Hvert neste tall er summen av de to forrige tallene.

La oss programmere starten på denne tallfølgen.

In [2]:
fib = [0, 1]

while len(fib) < 10:
    ny_fib = fib[-1] + fib[-2] # Tar summen av det siste og det nest siste tallet i listen.
    fib.append(ny_fib)         # Legger inn tallet i listen.

print(fib)

[0, 1, 1, 2, 3, 5, 8, 13, 21, 34]


Dette programmet legger inn det neste tallet i listen som summen av de to siste tallene så lenge listens lengde er under 10. Derfor skriver programmet ut de første ti *fibonaccitallene*.

---

#### Oppgaver

##### Oppgave 1

Lag en kalkulator som tar inn et areal for en sirkel og gir en radius som svar.

##### Oppgave 2

Hvis kvadrattallene er tallene med formel $n^2$ når $n\in \{1, 2, 3, \dots\}$ så må tallene med formel $n^3$ hete *kubikktallene*.

Skriv ut summen av de første $20$ kubikktallene og undersøk om $6859$ er et kubikktall.

##### Oppgave 3

Skriv ut de første 20 toerpotensene $2^n$. Har du sett disse tallene før?

##### Oppgave 4

> Se på koden for fibonaccitallene i eksempelet over.

*Det gylne snitt* er et viktig tall i matematikken. Hvis vi tar forholdet mellom et fibonaccitall og det forrige tallet, så vil verdien til dette forholdet bli mer og mer lik det gylne snitt. 

Finn en verdi for det gylne snitt ved å modifisere koden for fibonaccitallene. Søk etter en referanseverdi på nett for en fasit.

##### Oppgave 5

*Tribonaccitallene* er gitt ved følgende algoritme.

- Start med tallene $\{0, 1, 1\}$
- Det neste tallet er summen av de tre forrige tallene.

Skriv ut de første 20 tribonaccitallene.

##### Oppgave 6

Primtallene kjenner du til, ikke sant? Det er jo disse tallene som bare får et heltall som svar om man deler de på seg selv, eller $1$.

Skriv ut alle primtallene under $100$.